velox/serializers/PrestoSerializer.cpp

```diff
@@ -1733,6 +1733,13 @@ void PrestoVectorSerde::deserialize(
   VELOX_CHECK_EQ(
       checksum, actualCheckSum, "Received corrupted serialized page.");
 
+  if (type->children().empty()) {
```

Comment:
Would you add a test case?

Reply:
+1. Could you also comment why we need this handling? thanks!

Reply:
@xiaoxmeng  there is comments. Let me know what else you think is needed

Reply:
> AbstractTestNativeWindowQueries.testRowNumberWithFilter is the test that fails because of this bug. This PR fixes that test.

We cannot rely on Presto tests only. Please, add a unit test to Velox to illustrate this issue and verify it is fixed.